### PR TITLE
semantic/javascript:feature - adding try statement

### PR DESCRIPTION
**Pull request by nathanmartinszup**
Added try stament type to the genetic ast, walk validation to
the try statement and a parser for the javascript tree sitter
ast to the generic ast.

Signed-off-by: Nathan Martins <nathan.martins@zup.com.br>

**- What I did**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: internal/horusec-javascript/ast.go [hunk 1]
```diff
@@ -285,13 +286,34 @@ func (p *parser) parseStmt(node *cst.Node) ast.Stmt {
 			stmt.Else = p.parseStmt(alt)
 		}
 		return stmt
-	case ElseClause, StatementBlock:
+	case ElseClause, StatementBlock, FinallyClause:
```
Comment (matheusalcantarazup) [inline internal/horusec-javascript/ast.go:289]: I think that `FinallyClause` here is not necessary?

File: internal/ast/ast.go [hunk 2]
```diff
@@ -250,19 +250,36 @@ type (
...
+		Body        *BlockStmt
+		CatchClause []*CatchClause
+		Finalizer   *BlockStmt
+	}
+
+	CatchClause struct {
+		Position
+		Parameter *Ident
+		Body      *BlockStmt
+	}
```
Comment (matheusalcantarazup) [inline internal/ast/ast.go:265]: I think that would be better a documentation just clarifying that is maybe already claer.
```go
// TryStmt node represents a try catch finaly block.
type TryStmt struct {
	Position
	Body        *BlockStmt // Body of try.
	CatchClause []*CatchClause // List of catches or nil.
	Finalizer   *BlockStmt // Block of finaly statment or nil.
}

// CatchClause node represents a catch inside a try catch statement.
type CatchClause struct {
	Position
	Parameter *Ident // Parameter inside catch block.
	Body      *BlockStmt // Body of catch clause.
}
```

**Review by matheusalcantarazup — CHANGES_REQUESTED** [review]

File: internal/horusec-javascript/ast.go (hunk 1 above)
Comment (nathanmartinszup) [reply internal/horusec-javascript/ast.go]: Done!

File: internal/ast/ast.go (hunk 2 above)
Comment (nathanmartinszup) [reply internal/ast/ast.go]: Done!

File: internal/ast/ast.go [hunk 3]
```diff
@@ -250,19 +250,38 @@ type (
...
+		CatchClause []*CatchClause // List of catches or nil
+		Finalizer   *BlockStmt // Block of finally statement or nil
+	}
+
+	// CatchClause node represents a catch inside a try catch statement
+	CatchClause struct {
+		Position
+		Parameter *Ident // Parameter inside catch block
+		Body      *BlockStmt // Body of catch clause
+	}
```
Comment (matheusalcantarazup) [inline internal/ast/ast.go:267]: gofmt here?

**Review by matheusalcantarazup — CHANGES_REQUESTED** [review]

File: internal/ast/ast.go (hunk 3 above)
Comment (nathanmartinszup) [reply internal/ast/ast.go]: Done!

**Review by matheusalcantarazup — APPROVED** [review]

**Pull request merged**